crates/weaver_resolved_schema/src/v2/event.rs

```diff
@@ -0,0 +1,69 @@
+//! Event related definition structs.
+
+use schemars::JsonSchema;
+use serde::{Deserialize, Serialize};
+use weaver_semconv::{
+    attribute::RequirementLevel,
+    v2::{signal_id::SignalId, CommonFields},
+};
+
+use crate::v2::attribute::AttributeRef;
+
+/// The definition of an Event signal.
+#[derive(Serialize, Deserialize, Debug, Clone, PartialEq, JsonSchema)]
+#[serde(deny_unknown_fields)]
+pub struct Event {
+    /// The name of the event.
+    pub name: SignalId,
+
+    /// List of attributes that belong to this event.
+    #[serde(default)]
+    #[serde(skip_serializing_if = "Vec::is_empty")]
+    pub attributes: Vec<EventAttributeRef>,
+
+    // TODO - Should Entity Associations be "strong" links?
+    /// Which resources this span should be associated with.
```

Comment:
(and for other signals)

```suggestion
    /// Which entities this event should be associated with.
```